src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs

```diff
@@ -91,8 +91,15 @@ string searchToken
                                     OrderId = Guid.Parse(reader.GetString(7)),
                                     OppositeOrderId = Guid.Parse(reader.GetString(8)),
                                     TradeId = reader.GetString(9),
-                                    IsStraight = reader.GetString(1) == SearchToken // If the trade is straight or reverse.
-                                });
+                                    IsStraight =
+                                        reader.GetString(1) == SearchToken // If the trade is straight or reverse.
+                                };
+
+                                // We must ignore trades with negative prices and\or volumes (if any).
+                                if (trade.Price > 0 &&
+                                    trade.Volume > 0 &&
+                                    trade.OppositeVolume > 0)
+                                    result.Add(trade);
```

Comment:
Add warning loging and increment health indicator on else

Reply:
Implemented logging. But for being able to increment a healtch counter, we have to inject migration health service into the sql repository and make some else unwanted manipulations. Znd from the other side, our log will now contain all the worings about negative trades, which might be enough for analysis. Are you sure this healt update is actually needed?

Reply:
If it's hard to implement, we can omit this. But health statistics is much more easier to monitor. Warning message could be missed in the flow of messages.